samples/v2/add-and-drop-new-database-role.js

```diff
@@ -0,0 +1,102 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// sample-metadata:
+//  title: Add and drop new database role
+//  usage: node add-and-drop-new-database-role.js <INSTANCE_ID> <DATABASE_ID> <PROJECT_ID>
+
+'use strict';
+
+function main(
+  instanceId = 'my-instance',
+  databaseId = 'my-database',
+  projectId = 'my-project-id'
+) {
+  // [START spanner_add_and_drop_database_role]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+  // const projectId = 'my-project-id';
+
+  // Imports the Google Cloud client library
+  const {Spanner} = require('@google-cloud/spanner');
+
+  // creates a client
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  const databaseAdminClient = spanner.getDatabaseAdminClient();
+
+  async function addAndDropNewDatabaseRole() {
+    // Creates a new user defined role and grant permissions
+    try {
+      const request = [
+        'CREATE ROLE parent',
+        'GRANT SELECT ON TABLE Singers TO ROLE parent',
+        'CREATE ROLE child',
+        'GRANT ROLE parent TO ROLE child',
+      ];
+      const [operation] = await databaseAdminClient.updateDatabaseDdl({
+        database: databaseAdminClient.databasePath(
+          projectId,
+          instanceId,
+          databaseId
+        ),
+        statements: request,
+      });
+
+      console.log('Waiting for operation to complete...');
+      await operation.promise();
+
+      console.log('Created roles child and parent and granted privileges');
+    } catch (err) {
+      console.error('ERROR:', err);
+    }
+
+    // Revoke permissions and drop child role.
+    // A role can't be dropped until all its permissions are revoked.
+    try {
+      const request = ['REVOKE ROLE parent FROM ROLE child', 'DROP ROLE child'];
+      const [operation] = await databaseAdminClient.updateDatabaseDdl({
+        database: databaseAdminClient.databasePath(
+          projectId,
+          instanceId,
+          databaseId
+        ),
+        statements: request,
+      });
+
+      console.log('Waiting for operation to complete...');
+      await operation.promise();
+
+      console.log('Revoked privileges and dropped role child');
+    } catch (err) {
+      console.error('ERROR:', err);
+    } finally {
+      // Close the database when finished.
+      await databaseAdminClient.close();
```

Comment:
We shouldn't be closing the client directly right? This should be done via `spanner.close()`? How was it done in the old sample?

Reply:
In the old sample the database is simply getting closed as `database.close()`.  I have made the change from admin client closure to spanner client closure.